src/reactors/RelayOrderReactor.sol

```diff
@@ -1,46 +1,42 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity ^0.8.0;
 
-import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
-import {ERC20} from "solmate/src/tokens/ERC20.sol";
 import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
 import {Permit2Lib} from "permit2/src/libraries/Permit2Lib.sol";
-import {ISignatureTransfer} from "permit2/src/interfaces/ISignatureTransfer.sol";
-import {SignedOrder} from "UniswapX/src/base/ReactorStructs.sol";
 import {ReactorEvents} from "UniswapX/src/base/ReactorEvents.sol";
-import {CurrencyLibrary} from "UniswapX/src/lib/CurrencyLibrary.sol";
 import {IRelayOrderReactor} from "../interfaces/IRelayOrderReactor.sol";
-import {ResolvedRelayOrder, RelayOrder} from "../base/ReactorStructs.sol";
+import {ResolvedTransferDetails, RelayOrder} from "../base/ReactorStructs.sol";
 import {ReactorErrors} from "../base/ReactorErrors.sol";
 import {Multicall} from "../base/Multicall.sol";
 import {RelayOrderLib} from "../lib/RelayOrderLib.sol";
-import {ResolvedRelayOrderLib} from "../lib/ResolvedRelayOrderLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {ActionsLib} from "../lib/ActionsLib.sol";
+import {SignedOrder} from "UniswapX/src/base/ReactorStructs.sol";
 
 /// @notice Reactor for handling the execution of RelayOrders
 /// @notice This contract MUST NOT have approvals or priviledged access
 /// @notice any funds in this contract can be swept away by anyone
 contract RelayOrderReactor is Multicall, ReactorEvents, ReactorErrors, IRelayOrderReactor {
-    using SafeTransferLib for ERC20;
-    using CurrencyLibrary for address;
-    using ResolvedRelayOrderLib for ResolvedRelayOrder;
     using RelayOrderLib for RelayOrder;
-    /// @notice permit2 address used for token transfers and signature verification
+    using ActionsLib for bytes[];
 
+    /// @notice permit2 address used for token transfers and signature verification
     IPermit2 public immutable permit2;
+    /// @notice Actions only execute on the universal router.
     address public immutable universalRouter;
```

Comment:
btw why do we restrict to only Universal Router again? seems unnecessarily restrictive; and precludes this contract from being used for future features like gasless transfers

Reply:
We talked about opening it up.. I think its not out of the question/I dont see a problem with it

Reply:
Yea we could def open up in the future, this is for V1 right now. Also we can do gasless transfers, it would just be an order with no actions to be executed?

Reply:
Yeah we can do gasless transfers w p2